keyboards/planck/keymaps/a2b/keymap.c

```diff
@@ -0,0 +1,866 @@
+    KC_ESC,  _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, BEPO,    AZERTY,  _______, _______, DEBUG,
+    _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
+
+uint16_t hextokeycode(int hex) {
+    if (hex == 0x0) {
+        return KC_P0;
+    } else if (hex < 0xA) {
+        return KC_P1 + (hex - 0x1);
+    } else {
+        return KC_A + (hex - 0xA);
+    }
+}
+
+void send_alt_code(uint16_t altcode)
+{
+  mod_state = get_mods(); // save state of mods and delete Shift and Alt
+  del_mods(MOD_MASK_SA);
+  register_code(KC_LALT);
+  for (int i = 12; i >= 0; i -= 4) {
+    tap_code(hextokeycode((altcode >> i) & 0xF));
+  }
+  unregister_code(KC_LALT);
+  set_mods(mod_state); // put back the mods state
+}
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  if (IS_LAYER_ON(_AZERTY)) {
+    if (IS_LAYER_ON(_LOWER))  {
+      layer_on(_AZ_LOWER);
+    }
+    else {
+      layer_off(_AZ_LOWER);
+    }
+    if (IS_LAYER_ON(_RAISE))  {
+      layer_on(_AZ_RAISE);
+    }
+    else {
+      layer_off(_AZ_RAISE);
+    }
+  }
+  state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+  state = update_tri_layer_state(state, _SHIFT, _ALTGR, _SHALT);
+  state = update_tri_layer_state(state, _LOWER, _ALTGR, _LOWERAG);
+  state = update_tri_layer_state(state, _RAISE, _ALTGR, _RAISEAG);
+  return state;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+  case BEPO: // base layer, bépo layout using azerty driver on windows
+      if (record->event.pressed) {
+	set_single_persistent_default_layer(_BEPO);
+ 	layer_clear();
+      }
+      return false;
+      break;
+  case AZERTY: // base layer, bépo layout using azerty driver on windows
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_AZERTY);
+      }
+      return false;
+      break;
+  /* case AZ_LOWER: // turn on layer _SHIFT with modifier Left Shift held */
+  /*     if (record->event.pressed) { */
+  /*       layer_on(_LOWER); */
+  /*       layer_on(_AZ_LOWER); */
+  /*     } */
+  /*     else { */
+  /*       layer_off(_AZ_LOWER); */
+  /*       layer_off(_LOWER); */
+  /*     } */
+  /*     return false; */
+  /*     break; */
+  /* case AZ_RAISE: // turn on layer _SHIFT with modifier Left Shift held */
+  /*     if (record->event.pressed) { */
+  /*       layer_on(_RAISE); */
+  /*       layer_on(_AZ_RAISE); */
+  /*     } */
+  /*     else { */
+  /*       layer_off(_AZ_RAISE); */
+  /*       layer_off(_RAISE); */
+  /*     } */
+  /*     return false; */
+  /*     break; */
+  case SHIFT: // turn on layer _SHIFT with modifier Left Shift held
```

Comment:
the `LM` keycode will do this, for you. No need to have custom macros. 

Eg, `LM(_SHIFT, MOD_LSFT)` will do this.